src/Appium.Net/Appium/Android/KeyEvent.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using OpenQA.Selenium.Appium.Android.Enums;
+
+namespace OpenQA.Selenium.Appium.Android
+{
+    public class KeyEvent
+    {
+        private int? _keyCode;
+        private int? _metaState;
+        private int? _flags;
+
+        public KeyEvent()
+        {
+        }
+
+        /// <summary>
+        /// Creates a new key event
+        /// </summary>
+        /// <param name="keyCode">The key code. This is mandatory</param>
+        /// <see cref="AndroidKeyCode"/>
+        public KeyEvent(int keyCode)
+        {
+            _keyCode = keyCode;
+        }
+
+        /// <summary>
+        /// Sets the key code.
+        /// This is mandatory.
+        /// </summary>
+        /// <param name="keyCode">The key code</param>
+        /// <see cref="AndroidKeyCode"/>
+        /// <returns></returns>
+        public KeyEvent WithKeyCode(int keyCode)
+        {
+            _keyCode = keyCode;
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the meta key modifier
+        /// Flags indicating which meta keys are currently pressed.
+        /// Multiple meta key modifier flags can be combined into a single key event
+        /// </summary>
+        /// <param name="keyEventMetaModifier">The meta state</param>
+        /// <see cref="AndroidKeyMetastate"/>
+        /// <returns></returns>
+        public KeyEvent WithMetaKeyModifier(int keyEventMetaModifier)
+        {
+            _metaState |= keyEventMetaModifier;
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the flag(s).
+        /// Multiple flags can be combined into a single key event
+        /// </summary>
+        /// <param name="flag">The flag</param>
```

Comment:
are there also constants for flags?

Reply:
Annoyingly the're all in the AndroidKeyCode enum. I have added a reference to it